### DISCUSSION TITLE
Hadding headers in Route Middleware

### DISCUSSION THREAD

**Discussion by carbdias** [nuxt/nuxt#24330] [Questions] [answered]
### Describe the feature

Hello, I've tried all possible solution combinations, using 'to' variable, using ssrContext, using useRequestEvent composable, using the keywork .headers["field"] = value, and nothing works, every option gives a 500 error or don't change the header of the HTTP request at all. 

I also thought about creating a module with a hook, but I don't know how to connect it with the request or middleware.

Is there any viable option for this ?

Thank you in advance,

Note: This is the Content-language header, so I need to get the locale first, which cannot be done in nuxt.config file, so using routeRules is not an option for me.
Also, I cannot do it in server middleware because I cannot access to the useNuxtApp there an get the locale.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment, marked as answer]
Here's an example:

```ts
import { setHeader } from 'h3';

export default defineNuxtRouteMiddleware((to) => {
  if (import.meta.server) {
    setHeader(useRequestEvent(), 'x-client', 'set in route middleware');
  }
});

```

https://stackblitz.com/edit/github-sga62d?file=middleware/header-test.global.ts
